internal/cluster/clusterworkflow/activity_create_node_pool.go

```diff
@@ -38,25 +38,31 @@ const CreateNodePoolActivityName = "create-node-pool"
 type CreateNodePoolActivity struct {
 	clusters          cluster.Store
 	db                *gorm.DB
+	defaultVolumeSize int
 	nodePools         cluster.NodePoolStore
 	eksNodePools      eks.NodePoolStore
 	awsSessionFactory AWSSessionFactory
+	unsetVolumeSize   int
```

Comment:
It took me some time to figure out what this (`unset`) means. Is there a reason why we need that value injected here. I think it could just be a constant somewhere around `NewSelectVolumeSizeActivity`.

Reply:
My reasoning was that it is as much a configuration related information as the default one - we just want to alter what happens when to configuration is not exhaustive, so I tied it to the config and not the selection logic - the only difference is we wanted to give control of the default value to our users (pipeline operators) and we wanted to keep the unset value "internal" in the code so it is a development property. Along those lines I felt like it is best to keep the actual value of this close to the top level so it can be easily seen how this is implicitly part of the configuration of Pipeline and what inputs does the pipeline depend on rather than internalizing the value to the volume selector logic.

The only upside I consider right now for my approach is that in case we throw the volume selection logic out (which can be seen as a patch-only change), this approach makes it easier to notice that we have a special case handling for the configuration when a value is unset and less likely to accidentally regress to handling the unset case transparently and potentially changing how existing configurations behave in this case with newly created clusters - e.g. changing the unset default node volume size implicitly/potentially without notice.

But I'm going to internalize this to the volume selection activity then.

Reply:
[Fixed](https://github.com/banzaicloud/pipeline/compare/0e30f5dbe8ec96b1e168c848aadfd65cfc408226..4a93f8bc4925e894ff035584174682ce66d91d93).